Non-existent play folder causes build to fail

If the 'release-notes' folder does not exist the build will fail while doing the `publish` task to publish an apk/bundle.

Making `releaseNotesDir` in `PlayPublishPackageBase` be nullable again seems to make it work. It was intentionally removed here: https://github.com/Triple-T/gradle-play-publisher/commit/525b13203462e3417c68d5e3a8d158c5cf2d073d#diff-50be7e56abb0dd9419edb1f897d71a58

Sure, no worries :)

Config:
```
play {
    serviceAccountCredentials = file("../../signing/service-account.json")
    track = 'internal'
}
```

Error logs when running `./gradlew clean publishRelease --stacktrace`
```
> Task :app:publishReleaseApk FAILED

FAILURE: Build failed with an exception.

* What went wrong:
A problem was found with the configuration of task ':app:publishReleaseApk'.
> Directory '/Users/username/src/foo/app/build/generated/gpp/release/res/release-notes' specified for property 'releaseNotesDir$plugin' does not exist.

* Try:
Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Exception is:
org.gradle.api.tasks.TaskValidationException: A problem was found with the configuration of task ':app:publishReleaseApk'.
        at org.gradle.api.internal.tasks.TaskValidationContext$Severity$2.report(TaskValidationContext.java:55)
        at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:54)
        at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:101)
        at org.gradle.api.internal.tasks.execution.FinalizeInputFilePropertiesTaskExecuter.execute(FinalizeInputFilePropertiesTaskExecuter.java:44)
        at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:91)
        at org.gradle.api.internal.tasks.execution.ResolveTaskArtifactStateTaskExecuter.execute(ResolveTaskArtifactStateTaskExecuter.java:62)
        at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:59)
        at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:54)
        at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
        at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:34)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker$1.run(DefaultTaskGraphExecuter.java:256)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:317)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:309)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:185)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:97)
        at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.java:31)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:249)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:238)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:104)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:98)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.execute(DefaultTaskExecutionPlan.java:663)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.executeWithTask(DefaultTaskExecutionPlan.java:596)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.run(DefaultTaskPlanExecutor.java:98)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
        at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:46)
        at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
Caused by: org.gradle.api.InvalidUserDataException: Directory '/Users/username/src/foo/app/build/generated/gpp/release/res/release-notes' specified for property 'releaseNotesDir$plugin' does not exist.
        at org.gradle.api.internal.tasks.TaskValidationContext$Severity$2.report(TaskValidationContext.java:52)
        ... 25 more


* Get more help at https://help.gradle.org

BUILD FAILED in 28s
```

--- a/plugin/src/main/kotlin/com/github/triplet/gradle/play/internal/PlayPublishPackageBase.kt
+++ b/plugin/src/main/kotlin/com/github/triplet/gradle/play/internal/PlayPublishPackageBase.kt
@@ -18,7 +18,7 @@ abstract class PlayPublishPackageBase : PlayPublishTaskBase() {
     @get:PathSensitive(PathSensitivity.RELATIVE)
     @get:Optional
     @get:InputDirectory
-    internal val releaseNotesDir by lazy { File(resDir, RELEASE_NOTES_PATH) }
+    internal val releaseNotesDir by lazy { File(resDir, RELEASE_NOTES_PATH).safeMkdirs() }
 
     protected fun AndroidPublisher.Edits.updateTracks(editId: String, versions: List<Long>) {
         progressLogger.progress("Updating tracks")

--- a/plugin/src/main/kotlin/com/github/triplet/gradle/play/tasks/GenerateResources.kt
+++ b/plugin/src/main/kotlin/com/github/triplet/gradle/play/tasks/GenerateResources.kt
@@ -13,7 +13,6 @@ import com.github.triplet.gradle.play.internal.isDirectChildOf
 import com.github.triplet.gradle.play.internal.normalized
 import com.github.triplet.gradle.play.internal.nullOrFull
 import com.github.triplet.gradle.play.internal.orNull
-import com.github.triplet.gradle.play.internal.safeMkdirs
 import org.gradle.api.DefaultTask
 import org.gradle.api.tasks.CacheableTask
 import org.gradle.api.tasks.Internal
@@ -81,8 +80,6 @@ open class GenerateResources : DefaultTask() {
                         }
                     }
         }
-
-        ensureRootsExist()
     }
 
     private fun File.validate() {
@@ -119,18 +116,6 @@ open class GenerateResources : DefaultTask() {
 
         validateListings()
         validateReleaseNotes()
-    }
-
-    /**
-     * See https://github.com/gradle/gradle/issues/2016 to understand why this is necessary.
-     * Evaluation happens too early which means we either crash or our directories are ignored. To
-     * circumvent this issue, we simply make sure our inputs always exist.
-     */
-    private fun ensureRootsExist() = listOf(
-            LISTINGS_PATH,
-            RELEASE_NOTES_PATH
-    ).map { File(resDir, it) }.forEach {
-        it.safeMkdirs()
     }
 
     private fun File.findDest() = File(resDir, toRelativeString(findOwner()))

